### PR TITLE
Add utils::sanitize-entity tests

### PR DESCRIPTION
**Pull request by Convly**
Add tests for sanitize-entity.
Rise coverage to 100% and handle edge cases;

### CODE REVIEW

File: packages/strapi-utils/lib/__tests__/sanitize-entity.test.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        },
+      };
+
+      const { user: model } = models;
+
+      expect(sanitizeEntity(dataSource, { model })).toStrictEqual(
+        _.omit(input, ['email', 'password'])
+      );
+
+      expect(sanitizeEntity({ toJSON: () => null }, { model })).toBeNull();
```
Comment (petersg83) [inline packages/strapi-utils/lib/__tests__/sanitize-entity.test.js:279]: Why not use `jest.fn` and do `hasBeenCalled` ?

File: packages/strapi-utils/lib/__tests__/sanitize-entity.test.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      },
+    },
+  };
+
+  const userDzModel = {
+    ...userModel,
+    attributes: {
+      ...userModel.attributes,
+      dz: {
+        type: 'dynamiczone',
```
[the remark this answers is not in the document]
Comment (Convly) [reply packages/strapi-utils/lib/__tests__/sanitize-entity.test.js:74]: We could, but since we don't use them in the sanitation process, I didn't add them. Will do.

File: packages/strapi-utils/lib/__tests__/sanitize-entity.test.js (hunk 1 above)
Comment (Convly) [reply packages/strapi-utils/lib/__tests__/sanitize-entity.test.js:279]: Nice catch
